integration-tests/src/test_utils.rs

```diff
@@ -0,0 +1,634 @@
+        .unwrap();
+    }
+}
+
+/// End the poll. Advances the current height to expiry if necessary
+pub fn end_poll(app: &mut App, voting_verifier_address: &Addr, poll_id: PollID, expiry: u64) {
+    if app.block_info().height < expiry {
+        advance_to_height(app, expiry);
+    }
+    app.execute_contract(
+        Addr::unchecked("relayer"),
+        voting_verifier_address.clone(),
+        &voting_verifier::msg::ExecuteMsg::EndPoll { poll_id },
+        &[],
+    )
+    .unwrap();
+}
+
+pub fn construct_proof_and_sign(
+    app: &mut App,
+    messages: &[Message],
+    multisig_prover_address: &Addr,
+    multisig_address: &Addr,
+    workers: &Vec<Worker>,
+) -> Uint64 {
+    let res = app
+        .execute_contract(
+            Addr::unchecked("relayer"),
+            multisig_prover_address.clone(),
+            &multisig_prover::msg::ExecuteMsg::ConstructProof {
+                message_ids: messages.iter().map(|msg| msg.cc_id.to_string()).collect(),
+            },
+            &[],
+        )
+        .unwrap();
+
+    let msg_to_sign = get_event_attribute(&res.events, "wasm-signing_started", "msg")
+        .map(|attr| attr.value.clone())
+        .expect("couldn't find message to sign");
+    let session_id: Uint64 = get_event_attribute(&res.events, "wasm-signing_started", "session_id")
+        .map(|attr| attr.value.as_str().try_into().unwrap())
+        .expect("couldn't get session_id");
+
+    for worker in workers {
+        let signature = tofn::ecdsa::sign(
+            worker.key_pair.signing_key(),
+            &HexBinary::from_hex(&msg_to_sign)
+                .unwrap()
+                .as_slice()
+                .try_into()
+                .unwrap(),
+        )
+        .unwrap();
+
+        let sig = ecdsa::Signature::from_der(&signature).unwrap();
+
+        app.execute_contract(
+            worker.addr.clone(),
+            multisig_address.clone(),
+            &multisig::msg::ExecuteMsg::SubmitSignature {
+                session_id,
+                signature: HexBinary::from(sig.to_vec()),
+            },
+            &[],
+        )
+        .unwrap();
+    }
+    session_id
+}
+
+pub fn get_messages_from_gateway(
+    app: &mut App,
+    gateway_address: &Addr,
+    message_ids: &[CrossChainId],
+) -> Vec<Message> {
+    app.wrap()
+        .query_wasm_smart(
+            gateway_address,
+            &gateway::msg::QueryMsg::GetMessages {
+                message_ids: message_ids.to_owned(),
+            },
+        )
+        .unwrap()
+}
+
+pub fn get_proof(
+    app: &mut App,
+    multisig_prover_address: &Addr,
+    multisig_session_id: &Uint64,
+) -> multisig_prover::msg::GetProofResponse {
+    app.wrap()
+        .query_wasm_smart(
+            multisig_prover_address,
+            &multisig_prover::msg::QueryMsg::GetProof {
+                multisig_session_id: *multisig_session_id,
+            },
+        )
+        .unwrap()
+}
+
+pub fn advance_height(app: &mut App, num_blocks: u64) {
```

Comment:
```suggestion
pub fn advance_height(app: &mut App, offset: u64) {
```
or something better (increment?) if you can find it. I find num_blocks a bit too ambiguous, so I had to double-check the code so see if it was a replacement or addition